### PR TITLE
add nonce input. defaults to /dev/urandom

### PR DESCRIPTION
**Pull request by Jacoby6000**
```
  bigolchungus.sh [ -d <device id>         ]
                  [ -p <platform id>       ]
                  [ -l <local work size>   ]
                  [ -w <work set size      ]
                  [ -g <global work size>  ]
                  [ -k <kernel location>   ]
                  [ -n <hexadecimal nonce> ]
                  [ -v                     ]
                  <block>

  1. Device Selection

    -d <device id>
      Default `0`

    -p <platform id>
      Default `0`

    Run `clinfo -l` to get info about your device and platform ids.

  2. Open CL work configuration 

    -l <local work size> 
      Default `256`.

      If you are on AMD, `256` is probably the best value for you.
      If you are on nVidia, you probably want `1024`.

    -w <work set size> 
      Default `64`

    -g <global work size>
      Default `16777216` (1024 * 1024 * 16)

    -k <kernel location>
      If you are getting opencl error -46 or -30, try setting this to the absolute path of the `kernel.cl` file.
      Defaults to ./kernels/kernel.cl

  3. Debugging

    -v
      enable verbose mode.

  4. Advanced

    -n <hexadecimal nonce>
      Manually sets a nonce for hashing.
      In the unlikely case that your mining host provides a nonce, use this.
      If you are trying to get reproducable tests, use this.
```

### CODE REVIEW

**Comment by Jacoby6000** [comment]
```
 ✘  jbarber@home-dev  ~/projects/bigolchungus   feature/nonce-arg  test/test-speed.sh 3 -n beef -p 1 -- Configuring done                                                                         
-- Generating done
-- Build files have been written to: /home/jbarber/projects/bigolchungus
make: Entering directory '/home/jbarber/projects/bigolchungus'
make[1]: Entering directory '/home/jbarber/projects/bigolchungus'
make[2]: Entering directory '/home/jbarber/projects/bigolchungus'
make[2]: Leaving directory '/home/jbarber/projects/bigolchungus'
[100%] Built target bigolchungus
make[1]: Leaving directory '/home/jbarber/projects/bigolchungus'
make: Leaving directory '/home/jbarber/projects/bigolchungus'
Saving test data in file /home/jbarber/projects/bigolchungus/test/test-data-11-14-2019-180930

Iteration 1 Results so far: 955.38±0.00
Iteration 2 Results so far: 955.92±0.74
Iteration 3 Results so far: 956.33±0.82


Test complete.  Final results: 956.33±0.82
```

File: bigolchungus.cpp [hunk 1]
```diff
@@ -15,41 +16,41 @@
...
     "      If you are getting opencl error -46 or -30, try setting this to the absolute path of the `kernel.cl` file.\n"
     "      Defaults to ./kernels/kernel.cl\n\n"
     "  3. Debugging\n\n"
     "    -v\n"
     "      enable verbose mode.\n\n"
+    "  4. Advanced\n\n"
+    "    -n <hexadecimal nonce>\n"
+    "      Manually sets a nonce for hashing.\n"
+    "      In the unlikely case that your mining host provides a nonce, use this.\n"
+    "      If you are trying to get reproducable tests, use this.\n\n"
```
Comment (edmundnoble) [inline bigolchungus.cpp]: Typo

File: bigolchungus.cpp [hunk 2]
```diff
@@ -182,9 +189,15 @@ int main(int argc, char* const* argv) {
     // uint8_t* result = new uint8_t[nonce_step_size * 64];
 
     uint64_t start_nonce = 0;
-    FILE* urandom = fopen("/dev/urandom","rb");
-    fread(&start_nonce, 1, 8, urandom);
-    fclose(urandom);
+    if(nonceOverridden) {
```
Comment (edmundnoble) [inline bigolchungus.cpp]: All of the `if(` and `while(` stuff is ugly, fix that while you're in the area

File: kernels/kernel.cl [hunk 3]
```diff
@@ -6,8 +6,9 @@
 # define NATIVE_LITTLE_ENDIAN 1
 #endif
 
-#define BLAKE2S_BLOCKBYTES 64
+#define COMPARE_ALL
```
Comment (edmundnoble) [inline kernels/kernel.cl:9]: What's this mean, again

Comment (Jacoby6000) [reply kernels/kernel.cl:9]: See: https://github.com/kadena-community/bigolchungus/blob/c4a55f8824150daeb4f82d11e72285a5c151f575/kernels/kernel.cl#L269-L278

For some reason, running all of the comparisons gave me 0.5% faster hashing speed (955 -> 960 MH/s). We assume it has something to do with cache hits or something?

Comment (edmundnoble) [reply kernels/kernel.cl:9]: Lmao

**Pull request merged**